src/mesh/unstructured_mesh.C

```diff
@@ -993,6 +999,48 @@ void UnstructuredMesh::find_neighbors (const bool reset_remote_elements,
       }
   }
 
+#ifdef LIBMESH_ENABLE_PERIODIC
+  // Get the disconnected boundaries object (from periodic BCs)
+  auto * db = this->get_disconnected_boundaries();
+
+  if (db)
+    {
+      // Obtain a point locator
+      std::unique_ptr<PointLocatorBase> point_locator = this->sub_point_locator();
+
+      for (const auto & element : this->element_ptr_range())
+        {
+          for (auto ms : element->side_index_range())
+            {
+              // Skip if this side already has a valid neighbor (including remote neighbors)
+              if (element->neighbor_ptr(ms) != nullptr &&
+                  element->neighbor_ptr(ms) != remote_elem)
+                continue;
+
+              for (const auto & [id, boundary_ptr] : *db)
+                {
+                  if (!this->get_boundary_info().has_boundary_id(element, ms, id))
+                    continue;
+
+                  unsigned int neigh_side;
+                  const Elem * neigh =
+                    db->neighbor(id, *point_locator, element, ms, &neigh_side);
+
+                  if (neigh && neigh != remote_elem && neigh != element)
+                    {
+                      auto neigh_changeable = this->elem_ptr(neigh->id());
+                      element->set_neighbor(ms, neigh_changeable);
+                      neigh_changeable->set_neighbor(neigh_side, element);
+                    }
+                }
+            }
+        }
+
+      // Ghost the disconnected elements
+      this->add_ghosting_functor(std::make_unique<DisconnectedNeighborCoupling>(*this));
```

Comment:
No, this definitely isn't the way to do it.  It's better than requiring the user to handle adding a special ghosting functor, but if the user is modifying their mesh and relying on `find_neighbors` afterward, we don't want to be adding a new copy of a ghosting functor with each call.

Reply:
If we can just add the disconnected-neighbors check to the existing ghosting functor, then we're done.  If not ... I guess as a first pass we can do things this way, but at the very least `#ifdef LIBMESH_ENABLE_RTTI` we should be checking for an existing `DisconnectedNeighborCoupling` and not adding another one redundantly.